uk.co.compucorp.civicrm.hrleaveandabsences/js/test/shared/controllers/request.controller.spec.js

```diff
@@ -1342,33 +1346,57 @@
         });
       });
 
-      describe('user edits their own leave request popup', function () {
+      describe('when user edits their own leave request', function () {
         var leaveRequest;
 
-        ['staff', 'manager', 'admin'].forEach(function (permissionsRole) {
-          testRoleForSelfRecord(permissionsRole);
+        beforeEach(function () {
+          $rootScope.section = 'my-leave';
+          leaveRequest = LeaveRequestInstance.init();
         });
 
-        /**
-         * Tests the role for the self record and expects it to be "staff"
-         *
-         * @param {String} permissionsRole (staff|manager|admin)
-         */
-        function testRoleForSelfRecord (permissionsRole) {
-          describe('when user is ' + permissionsRole, function () {
+        describe('basic tests', function () {
+          beforeEach(function () {
+            spyOn(ContactInstance, 'checkIfSelfLeaveApprover').and.returnValue($q.resolve(false));
+          });
+
+          ['staff', 'manager', 'admin'].forEach(function (permissionsRole) {
             beforeEach(function () {
-              $rootScope.section = 'my-leave';
               role = permissionsRole;
-              leaveRequest = LeaveRequestInstance.init();
 
               initTestController({ leaveRequest: leaveRequest });
             });
 
             it('sets the staff role', function () {
-              expect(controller.isRole('staff')).toBeTruthy();
+              expect(controller.isRole('staff')).toBe(true);
             });
           });
-        }
+        });
+
+        describe('when user is a self leave approver', function () {
+          beforeEach(function () {
+            spyOn(ContactInstance, 'checkIfSelfLeaveApprover').and.returnValue($q.resolve(true));
+          });
+
+          ['staff', 'manager', 'admin'].forEach(function (permissionsRole) {
+            beforeEach(function () {
+              role = permissionsRole;
+
+              initTestController({ leaveRequest: leaveRequest });
+            });
+
+            it('sets the "admin" role', function () {
+              expect(controller.isRole('admin')).toBe(true);
+            });
+
+            it('sets the `isSelfLeaveApprover` public property to `true`', function () {
+              expect(controller.isSelfLeaveApprover).toBe(true);
```

Comment:
you need to check if `isSelfLeaveApprover` is set to false in the `basic tests` group`.

Reply:
Done.